lib/cli.js

```diff
@@ -98,7 +99,15 @@ if (options.debug) {
     debug('cli.input', cli.input);
 }
 
-npmCheck(options)
+(
```

Comment:
Personally not a fan of this style with a wide tenary and then chaining of that expression. What do you think about starting a Promise chain instead?

```js
Promise.resolve()
    .then(() => {
        return (options.installer === 'auto' ? detectPreferredInstaller(options.cwd) : options.installer)
    })
```

This will catch any potential synchrounous throws and move them to a rejected Promise, and thus to the `.catch` handler further down.